### ISSUE TITLE
[Bug]: Unable to view onchain wallet after update

### ISSUE THREAD

**Issue by bitcoinlizard** [btcpayserver/btcpayserver#4544]
### What is your BTCPay version?

1.7.4

### How did you deploy BTCPay Server?

manual

### What happened?

I am unable to look at my on chain wallets.  This seems to be a known bug (issue closed).  I only discovered the problem recently.  

I attempted this solution but it didn't work:
https://github.com/btcpayserver/btcpayserver/issues/4403#issuecomment-1347606557

### How did you encounter this bug?

Clicked on wallets > Bitcoin

I receive a blank screen

### Relevant log output

```shell
Jan 20 19:56:35 btcpay run.sh[1494]:   Exception data:
Jan 20 19:56:35 btcpay run.sh[1494]:     Severity: ERROR
Jan 20 19:56:35 btcpay run.sh[1494]:     SqlState: 42703
Jan 20 19:56:35 btcpay run.sh[1494]:     MessageText: column "AType" does not exist
Jan 20 19:56:35 btcpay run.sh[1494]:     Hint: Perhaps you meant to reference the column "wos.Type".
Jan 20 19:56:35 btcpay run.sh[1494]:     Position: 278
Jan 20 19:56:35 btcpay run.sh[1494]:     File: parse_relation.c
Jan 20 19:56:35 btcpay run.sh[1494]:     Line: 3633
Jan 20 19:56:35 btcpay run.sh[1494]:     Routine: errorMissingColumn
```

### What browser do you use?

Firefox 108.0.2

### Additional information

When I attempt to apply the DB fix posted posted by @NicolasDorier in the linked issue I receive this postgresql error:
[postgres@btcpay ~]$ psql -U btcpay -d btcpay
psql (14.6)
Type "help" for help.

btcpay=> UPDATE  nbxv1_metadata SET data='"Descriptors"' WHERE key='ImportAddressToRPC' AND data != '"False"';
ERROR:  relation "nbxv1_metadata" does not exist
LINE 1: UPDATE  nbxv1_metadata SET data='"Descriptors"' WHERE key='I...

### Are you sure this is a bug report?

- [X] I confirm this is a bug report

**Comment by bitcoinlizard** [comment]
I see that I was trying to run this on the incorrect DB.  I ran the it against the nbxplorer DB:

[postgres@btcpay ~]$ psql -U nbxplorer -d nbxplorer
psql (14.6)
Type "help" for help.

nbxplorer=> UPDATE  nbxv1_metadata SET data='"Descriptors"' WHERE key='ImportAddressToRPC' AND data != '"False"';
UPDATE 0

Unfortunately the result is the same.  I get a blank screen when trying to access either of my on chain wallets.

**Comment by bitcoinlizard** [comment]
Additional details.  This is actually a "HTTP ERROR 500" error and not a blank screen.  Firefox displays it as a blank screen for some reason but Chrome gives the 500 error.

I have 2 stores and both have hot wallets that were created in BTCPay server.  I believe only one of them has transactions in it, the other has not been used.  They are both currently empty.

I would be fine with dumping these wallets to get the on chain functionality working however I do believe this is a bug.  Although this is a manually built system I have only ever used the tagged releases with this Arch Linux system.  I update the system about every month and being Arch Linux it is using generally using very up to date packages.  For some reason the on chain wallet stopped working in the last couple months.

This is a VM so I could snap shot it and try downgrading if that would be a means of resolving the issue.

**Comment by NicolasDorier** [comment]
The problem you have is because you ran btcpay from master branch rather than from a released version.

The fix you attempted with `UPDATE nbxv1_metadata SET data='"Descriptors"' WHERE key='ImportAddressToRPC' AND data != '"False"';` has nothing to do with your problem.

The problem is that your DB has column that never existed in released version.

**Comment by bitcoinlizard** [comment]
Thank you for getting back to me.  

I believe I have only ever used tagged releases.  I used the Extended Manual guide to build my server.  I have only ever used these commands to perform upgrades to nbxplorer and btcpay server as per the guide instructions:

cd ~; pushd ~/src/NBXplorer; git pull; ./build.sh; popd;
cd ~; pushd ~/src/btcpayserver; git pull; ./build.sh; popd;

I assume those only use tagged releases?

Other thing to note that is maybe relevant and I forgot to mention.  I've been running this server for several years and originally Nbxplorer was using the DBTrie DB and maybe 6 months or so ago I moved it to Postgres per recommendations on github.  I am pretty sure the on chain wallet worked following that change.  The little graph on the Dashboard that relies on Nbxplorer running Postgres is still working.

I usually upgrade when there is a new release but not every time.  I might have jumped one of the 1.7.X releases.  Not sure if they are cumulative (probably) or if perhaps there was a DB change in one of them that I missed by skipping a release.

If it's too much trouble I can dump these on chain wallets and start fresh.  Not sure if there is a guide to doing so, I assume I have to delete the data in a table such as "nbxv1_metadata"?

**Comment by bitcoinlizard** [comment]
I ended up dropping the btcpay and nbxplorer databases and setting up my BTCpay server fresh like a new install.  It was very fast to set it up and the on-chain wallet now works correctly.

I would recommend updating the documentation for the extended manual setup to use tagged versions of btcpay and nbxplorer:
https://docs.btcpayserver.org/Deployment/ManualDeploymentExtended/

Currently the install and update commands will just pull from master:
cd ~; pushd ~/src/NBXplorer; git pull; ./build.sh; popd;
cd ~; pushd ~/src/btcpayserver; git pull; ./build.sh; popd;

@NicolasDorier mentioned that using master is not a good idea and in my case resulted in a broken on-chain wallet.

This ticket can be closed as my issue has been resolved.  I just wanted a responsible party to see my suggestion about updating the documentation so this doesn't happen to someone else in the future.
